autoscheduler/frontend/src/tests/ui/SelectTerm.test.tsx

```diff
@@ -12,60 +18,85 @@ jest.mock('@reach/router', () => ({
   navigate: jest.fn(),
 }));
 
+// Mocks the fetch call to api/terms
+const mockTermsAPI = (): void => {
+  fetchMock.mockResponseOnce(JSON.stringify({
+    'Fall 2020': '202031',
+    'Summer 2020': '202021',
+    'Spring 2020': '202011',
+    'Fall 2019': '201931',
+    'Summer 2019': '201921',
+    'Spring 2019': '201911',
+  }));
+};
+
+
 describe('SelectTerm', () => {
+  beforeEach(mockTermsAPI);
+
+  afterEach(fetchMock.mockRestore);
+
   describe('Menu opens', () => {
-    test('Menu opens after button is clicked', () => {
+    test('Menu opens after button is clicked', async () => {
       // arrange
       const store = createStore(autoSchedulerReducer);
 
-      const { getByText } = render(
-        <Provider store={store}>
-          <SelectTerm />
-        </Provider>,
-      );
+      let getByText: Function;
+      await act(async () => {
+        ({ getByText } = render(
+          <Provider store={store}>
+            <SelectTerm />
+          </Provider>,
+        ));
+      });
 
       // act
       const button = getByText('Select Term');
-      fireEvent.click(button);
+      act(() => { fireEvent.click(button); });
 
       // assert
       expect(document.getElementsByClassName('MuiPopover-root')[0]).not.toHaveAttribute('aria-hidden');
     });
   });
 
   describe('Menu is closed', () => {
-    test('on initialization', () => {
+    test('on initialization', async () => {
       // arrange/act
       const store = createStore(autoSchedulerReducer);
 
-      render(
-        <Provider store={store}>
-          <SelectTerm />
-        </Provider>,
-      );
+      await act(async () => {
+        render(
+          <Provider store={store}>
+            <SelectTerm />
+          </Provider>,
+        );
+      });
 
       // assert
       expect(document.getElementsByClassName('MuiPopover-root')[0]).toHaveAttribute('aria-hidden');
     });
   });
 
   describe('Redirects to /schedule', () => {
-    test('When term is selected', () => {
+    test('When term is selected', async () => {
       // arrange
       const store = createStore(autoSchedulerReducer);
 
-      const { getByText } = render(
-        <Provider store={store}>
-          <SelectTerm />
-        </Provider>,
-      );
-
+      let getByText: Function;
+      await act(async () => {
```

Comment:
The way that I've been doing this is, instead of wrapping the render in `act`, I swap `getByText` for `findByText`, which you can use below with an `await`. I think that reads a little better, if it works